client/pkg/transport/listener.go

```diff
@@ -377,18 +381,32 @@ func (info TLSInfo) baseConfig() (*tls.Config, error) {
 	// Client certificates may be verified by either an exact match on the CN,
 	// or a more general check of the CN and SANs.
 	var verifyCertificate func(*x509.Certificate) bool
-	if info.AllowedCN != "" {
-		if info.AllowedHostname != "" {
-			return nil, fmt.Errorf("AllowedCN and AllowedHostname are mutually exclusive (cn=%q, hostname=%q)", info.AllowedCN, info.AllowedHostname)
+	var definedRestrictions int
+	for _, restriction := range []string{info.AllowedCN, info.AllowedHostname, info.AllowedURI} {
+		if restriction != "" {
+			definedRestrictions++
 		}
+	}
+	switch {
+	case definedRestrictions > 1:
+		return nil, errors.New("exactly one of AllowedCN, AllowedHostname, or AllowedURI can be defined")
```

Comment:
I think error message like `exactly one of --peer-cert-allowed-cn, --peer-cert-allowed-hostname, or --peer-cert-allowed-uri can be defined` might be clearer for users.

Reply:
This might be confusing for embedded etcd users since they would not be interfacing with any command line parameters. I'm happy to change it either way though.

Reply:
@LINKIWI got it, then I think it's fine